tests/functional/spec/features/submit_page/submit_page_summary.spec.js

```diff
@@ -0,0 +1,100 @@
+import SubmitPage from "../../../generated_pages/summary/submit.page.js";
+
+describe("Submit Page with Summary", () => {
+  beforeEach("Load the survey", () => {
+    browser.openQuestionnaire("test_summary.json");
+  });
+
+  it("Given a survey with a summary has been completed when the submit page is displayed, then it should contain a summary of all answers", () => {
+    completeAllQuestions();
+
+    expect($(SubmitPage.radioAnswer()).getText()).to.contain("Bacon");
+    expect($(SubmitPage.dessertGroupTitle()).getText()).to.contain("Dessert");
+    expect($(SubmitPage.dessertAnswer()).getText()).to.contain("Crème Brûlée");
+    expect($(SubmitPage.dessertConfirmationAnswer()).getText()).to.contain("Yes");
+    expect($(SubmitPage.numbersCurrencyAnswer()).getText()).to.contain("£1,234.00");
+    expect($(SubmitPage.numbersUnitAnswer()).getText()).to.contain("123,456 km²");
+    expect($(SubmitPage.numbersDecimalAnswer()).getText()).to.contain("123,456.78");
+  });
+
+  it("Given a survey with a summary has been completed when the submit page is displayed then I should be able to submit the answers", () => {
+    completeAllQuestions();
+
+    $(SubmitPage.submit()).click();
+    expect(browser.getUrl()).to.contain("thank-you");
+  });
+
+  it("Given a survey with a summary has been completed when a summary page edit link is clicked then it should return to that question", () => {
+    completeAllQuestions();
+
+    $(SubmitPage.radioAnswerEdit()).click();
+
+    expect($(RadioPage.bacon()).isSelected()).to.be.true;
+  });
+
+  it("Given a survey with a summary has been completed when a summary page edit link is clicked then it should return to that question then back to summary", () => {
+    completeAllQuestions();
+
+    $(SubmitPage.radioAnswerEdit()).click();
+    $(RadioPage.sausage()).click();
+    $(RadioPage.submit()).click();
+    expect($(SubmitPage.radioAnswer()).getText()).to.contain("Sausage");
+  });
+
+  it("Given the edit link is used when a question is updated then the submit page summary should show the new answer", () => {
+    completeAllQuestions();
+
+    $(SubmitPage.numbersUnitAnswerEdit()).click();
+    expect($(NumbersPage.unit()).isFocused()).to.be.true;
+    $(NumbersPage.unit()).setValue("654321");
+    $(NumbersPage.submit()).click();
+    expect($(SubmitPage.numbersUnitAnswer()).getText()).to.contain("654,321 km²");
+  });
+
+  it("Given a number value of zero is entered when on the submit page then formatted 0 should be displayed on the summary", () => {
+    $(RadioPage.submit()).click();
+    $(DessertPage.answer()).setValue("Cake");
+    $(DessertPage.submit()).click();
+    $(DessertConfirmationPage.yes()).click();
+    $(DessertConfirmationPage.submit()).click();
+    $(NumbersPage.currency()).setValue("0");
+    $(NumbersPage.submit()).click();
+    expect($(SubmitPage.numbersCurrencyAnswer()).getText()).to.contain("£0.00");
+  });
```

Comment:
This should really be a test for the currency answer type.

Reply:
I have left it as is for now. A currency spec didn't exist and didn't want to introduce a new currency spec with just this.